internal/internal_command_state_machine.go

```diff
@@ -1084,11 +1093,17 @@ func (h *commandsHelper) recordVersionMarker(changeID string, version Version, d
 		panic(err)
 	}
 
+	searchAttributeWasUpdatedPayload, err := dc.ToPayloads(searchAttributeWasUpdated)
+	if err != nil {
+		panic(err)
+	}
+
 	recordMarker := &commandpb.RecordMarkerCommandAttributes{
 		MarkerName: versionMarkerName,
 		Details: map[string]*commonpb.Payloads{
-			versionMarkerChangeIDName: changeIDPayload,
-			versionMarkerDataName:     versionPayload,
+			versionMarkerChangeIDName:         changeIDPayload,
+			versionMarkerDataName:             versionPayload,
+			versionSearchAttributeUpdatedName: searchAttributeWasUpdatedPayload,
```

Comment:
I need to adjust this to not send if it is not false. 

This flag isn't strictly needed but because the Go SDK counts command IDs and processes history out of order it is useful to have to avoid refactoring how the SDK replays commands  just for this edge case.